app/controllers/tag_controller.rb

```diff
@@ -43,6 +43,8 @@ def index
   def show
     # try for a matching /wiki/_TAGNAME_ or /_TAGNAME_
     @wiki = Node.where(path: "/wiki/#{params[:id]}").try(:first) || Node.where(path: "/#{params[:id]}").try(:first)
+    @wiki = Node.find(@wiki.power_tag('redirect'))  if @wiki.has_powertag('redirect') # use a redirected wiki page if it exists
```

Comment:
OK, so maybe occasionally there is no @wiki variable and we can't run `.has_powertag` on `nil` -- so can we add:

`if @wiki && @wiki.has_powertag('redirect')` to catch this?

Reply:
oof, that one almost stumped me!